lib/action_policy/testing.rb

```diff
@@ -34,21 +34,23 @@ def context_matches?(context, actual)
       end
 
       class Scoping # :nodoc:
-        attr_reader :policy, :target, :type, :name, :scope_options
+        attr_reader :policy, :target, :type, :name, :scope_options, :context
 
         def initialize(policy, target, type, name, scope_options)
           @policy = policy
           @target = target
           @type = type
           @name = name
           @scope_options = scope_options
+          @context = policy.authorization_context
         end
 
-        def matches?(policy_class, actual_type, actual_name, actual_scope_options)
+        def matches?(policy_class, actual_type, actual_name, actual_scope_options, actual_context)
           policy_class == policy.class &&
             type == actual_type &&
             name == actual_name &&
-            actual_scope_options === scope_options
+            actual_scope_options === scope_options &&
+            actual_context.all? { |key, value| context[key] == value }
```

Comment:
We should use the same logic here as for authorization calls: https://github.com/palkan/action_policy/blob/8e6923970a713bde164680df920522e98fff0d9b/lib/action_policy/testing.rb#L29-L33

It covers subsets and RSpec composable matchers.